daemon/CryptoImpl/cryptopluginfunctionwrappers.cpp

```diff
@@ -41,67 +41,75 @@ bool CryptoPluginWrapper::setLockCode(
 }
 
 DataResult CryptoPluginWrapper::generateRandomData(
-        CryptoPlugin *plugin,
+        const PluginAndCustomParams &pluginAndCustomParams,
```

Comment:
The problem is that QtConcurrent::run() only works for functions that take five or less arguments, so in some cases I had to compress stuff.  It's ugly as sin, and I agree it's bad.  But it's internal detail which we can make nicer later if required.

The reason why I join the plugin and custom parameters together is because the parameters really are plugin specific, so it "kind of" makes sense.

Reply:
Ok 🙂